### PR TITLE
fix: hidden attribute implies display:none

### LINKED ISSUE #3111: window.getComputedStyle() on hidden element
### Basic info:

- **jsdom version:** 16.4

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`
      <div id="foo" hidden>foo</div>
      <div id="bar" hidden style="display: block;">bar</div>
`);

console.log(window.getComputedStyle(document.getElementById("foo")).display)
```

The computed style for `#foo` is `display: block` in JSDOM.

https://codesandbox.io/s/getcomputedstyle-on-hidden-uhtqm?file=/src/index.test.js

### How does similar code behave in browsers?

The computed style for elements with `hidden` attribute is `display: none` if it is not overwritten by a stylesheet.

https://codesandbox.io/s/getcomputedstyle-on-hidden-uhtqm?file=/src/index.js

### PR DESCRIPTION
**Pull request by ph-fritsche**
As I haven't worked on this repo before, I'm not sure if I targeted the correct files with the changes.
Test script failed on setup on my machine. I hope the CI will verify if the changes do work.

The changes should fix #3111 

The [`hidden` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) implies `display: none` for the element which can be overwritte by any style rule.

### CODE REVIEW

**Comment by ph-fritsche** [comment]
@domenic Could you have a look at this? The CI is pending forever. There seems to be something wrong with the setup.

**Comment by vegardbb** [comment]
@ph-fritsche The Travis subscription has been terminated. Hence, the CI setup defined in travis.yml does not run.

**Comment by ph-fritsche** [comment]
@vegardbb How should we procede here? Is there a replacement (Github Actions?) in the making or will a maintainer verify this by other means?
We've got a fix at [testing-library/user-event](https://github.com/testing-library/user-event/pull/556) that requires this bug to be fixed first.

**Comment by vegardbb** [comment]
@ph-fritsche Try amending your last commit (change the message or something) or make a dummy commit. The CI should be running then.

**Comment by ph-fritsche** [comment]
@vegardbb Those tests not timing out are passing

**Comment by vegardbb** [comment]
@ph-fritsche You might want to look into this error, which is thrown across all the builds. Briefly described, the test suite had the file "mapped-attribute-adopt-001.html" marked as a failing test. All you have to do is remove that entry from the file ["to-run.yaml"](https://github.com/ph-fritsche/jsdom/blob/fix-3111/test/web-platform-tests/to-run.yaml#L460). As for the other errors thrown in the build, I am already looking into them in another pull request.
```
  1) web-platform-tests
       html/dom/elements/global-attributes
         [expected fail] mapped-attribute-adopt-001.html:
     Error: 
            Hey, did you fix a bug? This test used to be failing, but during
            this run there were no errors. If you have fixed the issue covered
            by this test, you can edit the "to-run.yaml" file and remove the line
            containing this test. Thanks!
            
      at /home/runner/work/jsdom/jsdom/test/web-platform-tests/run-single-wpt.js:190:22
      at http://web-platform.test:9000/resources/testharness.js:2950:22
      at forEach (http://web-platform.test:9000/resources/testharness.js:3653:26)
      at Tests.notify_complete (http://web-platform.test:9000/resources/testharness.js:2947:9)
      at all_complete (http://web-platform.test:9000/resources/testharness.js:2817:22)
      at Tests.complete (http://web-platform.test:9000/resources/testharness.js:2829:13)
      at http://web-platform.test:9000/resources/testharness.js:2620:26
      at callTheUserObjectsOperation (lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (lib/jsdom/living/events/EventTarget-impl.js:318:25)
      at invokeEventListeners (lib/jsdom/living/events/EventTarget-impl.js:274:3)
      at EventTargetImpl._dispatch (lib/jsdom/living/events/EventTarget-impl.js:221:9)
      at fireAnEvent (lib/jsdom/living/helpers/events.js:18:36)
      at Document.<anonymous> (lib/jsdom/browser/Window.js:776:9)
      at Document.callTheUserObjectsOperation (lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (lib/jsdom/living/events/EventTarget-impl.js:318:25)
      at invokeEventListeners (lib/jsdom/living/events/EventTarget-impl.js:274:3)
      at DocumentImpl._dispatch (lib/jsdom/living/events/EventTarget-impl.js:221:9)
      at fireAnEvent (lib/jsdom/living/helpers/events.js:18:36)
      at dispatchEvent (lib/jsdom/living/nodes/Document-impl.js:475:9)
      at /home/runner/work/jsdom/jsdom/lib/jsdom/living/nodes/Document-impl.js:480:11
      at new Promise (<anonymous>)
      at onLoad (lib/jsdom/living/nodes/Document-impl.js:478:14)
      at Object.check (lib/jsdom/browser/resources/resource-queue.js:76:23)
      at /home/runner/work/jsdom/jsdom/lib/jsdom/browser/resources/resource-queue.js:83:27
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
```

**Comment by ph-fritsche** [comment]
@vegardbb @domenic Is there a timeframe for the missing CI jobs (node 14 and 15) to be fixed? Or could someone verify that the test failures are not caused by the changes here and merge this earlier so that we could start applying our bugfix in `user-event`?

**Comment by domenic** [comment]
There is no timeframe; the maintainers work on this project in their spare time, and I at least have been experiencing significant burnout on open-source work since COVID. We're doing the best we can.

We won't be merging pull requests while master has non-green CI, so getting that fixed is the first priority.

**Comment by ph-fritsche** [comment]
@domenic The tests pass now. :rocket: 
Thanks to you and @vegardbb :)

File: test/web-platform-tests/to-upstream/cssom/getComputedStyle-hidden.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/cssom/getComputedStyle-hidden.html:1]: Can you move this test to a new `html/rendering/the-css-user-agent-style-sheet-and-presentational-hints` folder (undernearth to-upstream)?

**Review by domenic — COMMENTED** [review]
Looking good; one minor nit to fix.

**Comment by ph-fritsche** [comment]
When will the release containing these changes be published?

**Comment by domenic** [comment]
Same answer as https://github.com/jsdom/jsdom/pull/3112#issuecomment-779881079

**Comment by ph-fritsche** [comment]
I know you said you're not interested [in the other PR](https://github.com/jsdom/jsdom/pull/3115).
But automating tasks like releases from master helps the maintainers and motivates contributors. ;)

If work needs to be done, it's perfectly understandable that it takes some time.
And to ask someone to pick up some of the work if they want to speed it up, is perfectly fine.
But "it's my release button and I press it when I press it - today or next year - you'll see when it happens" is detrimental to the thought that collaborating on open source benefits everbody.

**Pull request merged**